src/app/core/common-components/entity-subrecord/entity-subrecord/entity-subrecord.component.ts

```diff
@@ -96,6 +98,8 @@ export class EntitySubrecordComponent<T extends Entity> implements OnChanges {
   @Input() isLoading: boolean;
 
   @Input() clickMode: "popup" | "navigate" | "none" = "popup";
+  
+  @Output() selectedRecords: EventEmitter<{ event: any, row: T[] }> = new EventEmitter<{ event: any, row: T[] }>
```

Comment:
```suggestion
  /** outputs an event containing an array of currently selected records (checkmarked by the user) */
  @Output() selectedRecords: EventEmitter<T[]> = new EventEmitter<T[]>
```

I would like to simplify this interface: Let's simply output the array of all currently selected records, whenever the selection changes. I don't think any component outside of this will need access to individual select and unselect events.

That'll let you simplify the code in EntityList, but requires you to adapt selectRow() here

Reply:
Changed it.